extensions/amp-script/0.1/user-activation-tracker.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+export const ACTIVATION_TIMEOUT = 5000; // 5 seconds.
+
+const ACTIVATION_EVENTS = ['click', 'input', 'dblclick', 'keypress', 'submit'];
+
+
+/**
+ * See https://github.com/dtapuska/useractivation for inspiration.
```

Comment:
The final spec is a bit different https://mustaqahmed.github.io/user-activation-v2/  and the names are:
`HasSeenUserActivation` and `HasConsumableUserActivation` instead of `hasBeenActive` and `isActive`.

(also a big [change](https://docs.google.com/document/d/1NKLJ2MBa9lA_FKRgD2ZIO7vIftOJ_YiXXMYfRMdlV-s/edit?ts=5c7d1ca1#) in the final spec is that **both** of these bits get **reset** as soon as it is delegated to another frame via `postMessage`)

Reply:
@aghassemi For now this is strictly `amp-script` use cases, so no consumption model quite yet, but likely to show up. I'm not sure we ever publish it to other components that want to polyfill this API, though I think I'd like to. But there's definitely more to come in this area.